### PR TITLE
JDK-8075909: [TEST_BUG] The regression-swing case failed as it does not have the 'Open' button when select 'subdir' folder with NimbusLAF

### PR DESCRIPTION
**Pull request by amresh-sahu**
Test name(s): javax/swing/JFileChooser/6698013/bug6698013.java

Problem: A manual regression test case instructions does not match with the UI shown to the user. Instruction says to click Open button while the File chooser dialog shows the Save button.

Fix: There is code in test case to open file save dialog instead of file open dialog. Test case modified to show open file dialog. Fix has been tested on windows platform. This issue is not OS specific.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8075909](https://bugs.openjdk.java.net/browse/JDK-8075909): [TEST_BUG] The regression-swing case failed as it does not have the 'Open' button when select 'subdir' folder with NimbusLAF

### Reviewers
 * [Alexey Ivanov](https://openjdk.java.net/census#aivanov) (@aivanov-jdk - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2587/head:pull/2587`
`$ git checkout pull/2587`

### CODE REVIEW

**Comment by amresh-sahu** [comment]
@shurymury @akolarkunnu Please review my changes.

File: test/jdk/javax/swing/JFileChooser/6698013/bug6698013.java [hunk 1]
```diff
@@ -45,13 +45,13 @@
...
+        chooser.showOpenDialog(null);
     }
 
     public void init() {
         JFileChooser chooser = new JFileChooser(new VirtualFileSystemView());
         chooser.setCurrentDirectory(root);
-        chooser.showSaveDialog(null);
+        chooser.showOpenDialog(null);
     }
 }
```
Comment (aivanov-jdk) [inline test/jdk/javax/swing/JFileChooser/6698013/bug6698013.java:56]: Shall we eliminate code duplication between main() and init()?

Comment (amresh-sahu) [reply test/jdk/javax/swing/JFileChooser/6698013/bug6698013.java:56]: init method is part of applet version of test case. It needs to be completely converted to java version. I will do the appropriate changes.

Comment (aivanov-jdk) [reply test/jdk/javax/swing/JFileChooser/6698013/bug6698013.java:56]: > 
> 
> init method is part of applet version of test case. It needs to be completely converted to java version. I will do the appropriate changes.

No, it's not necessary. I meant the opposite. You can create an instance of the test class and call init():
```
SwingUtilities.invokeAndWait(() -> new bug6698013().init());
```
Moreover, Swing components should be created and manipulated on EDT, which does not hold for the current main().

Comment (amresh-sahu) [reply test/jdk/javax/swing/JFileChooser/6698013/bug6698013.java:56]: I will incorporate the suggested change.

File: test/jdk/javax/swing/JFileChooser/6698013/bug6698013.java [hunk 2]
```diff
@@ -43,9 +44,11 @@
     final static VirtualFile subdirFile = new VirtualFile("testdir/subdir/subtest.txt", false);
 
     public static void main(String[] args) {
-        JFileChooser chooser = new JFileChooser(new VirtualFileSystemView());
-        chooser.setCurrentDirectory(root);
-        chooser.showOpenDialog(null);
+        try {
+            SwingUtilities.invokeAndWait(() -> new bug6698013().init());
+        } catch (InvocationTargetException | InterruptedException e) {
+           e.printStackTrace();
```
Comment (shurymury) [inline test/jdk/javax/swing/JFileChooser/6698013/bug6698013.java:50]: The test should, probably, fail in case of the exception.

Comment (aivanov-jdk) [reply test/jdk/javax/swing/JFileChooser/6698013/bug6698013.java:50]: Yes, it's safe to declare `main` with `throws Exception`; `main` is for convenience to run the test quickly on the command line.

The test harness does not use `main`, it runs the test as applet.

**Review by prsadhuk — COMMENTED** [review]
I guess since you are modifying the test, it's better to convert into main based  from applet.
Please remove %W% %E% from @test tag and also @author tag
Also, JFileChooser.showOpenDialog() can throw HeadlessException, so you need to mark the test @headful once you convert to main based.

**Comment by aivanov-jdk** [comment]
> 
> 
> I guess since you are modifying the test, it's better to convert into main based from applet.
> Please remove %W% %E% from @test tag and also @author tag
> Also, JFileChooser.showOpenDialog() can throw HeadlessException, so you need to mark the test @headful once you convert to main based.

@mrserb is against converting manual applet-based tests one-by-one, see [his comment](https://github.com/openjdk/jdk/pull/2094#discussion_r558717221) in #2094. Thus, I'd rather leave the test as is. However, I agree `%W% %E%` can be removed from `@test` tag and `@author` tag can be dropped completely.

File: test/jdk/javax/swing/JFileChooser/6698013/bug6698013.java [hunk 3]
```diff
@@ -43,12 +43,8 @@
 
     final static VirtualFile subdirFile = new VirtualFile("testdir/subdir/subtest.txt", false);
 
-    public static void main(String[] args) {
-        try {
-            SwingUtilities.invokeAndWait(() -> new bug6698013().init());
-        } catch (InvocationTargetException | InterruptedException e) {
-           e.printStackTrace();
-        }
+    public static void main(String[] args) throws InvocationTargetException, InterruptedException {
```
Comment (aivanov-jdk) [inline test/jdk/javax/swing/JFileChooser/6698013/bug6698013.java:46]:
```suggestion
    public static void main(String[] args) throws Exception {
```
For simplicity.

**Comment by aivanov-jdk** [comment]
I would also suggest expanding the wildcard import
```
import javax.swing.*;
```
with imports of specific classes that are used by the test and then sorting the imports. An IDE can do this for you.

**Review by aivanov-jdk — APPROVED** [review]

**Pull request closed without merging**